### PR TITLE
PLAT-96 Fixed Sorting and Filtering of Transient Fields

### PR DESCRIPTION
**Pull request by oliver-richers**
- PLAT-96 TransientFieldValueType Now Considers Natural Order Comparator
- PLAT-96 Renamed IEmfDataTableStrategy to IEmfAttributeDataTableStrategy
- PLAT-96 Removed Unnecessary Code
- PLAT-96 Removed EmfForeignEntityPlainListAttribute
- PLAT-96 Moved EmfAttributeColumnHandler
- PLAT-96 Added New EmfByteArrayAttribute and Removed EmfDummyAttribute
- PLAT-96 Removed IEmfAttributeDataTableStrategy and All Related Code

### CODE REVIEW

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/transients/TransientFieldValueType.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
 	public TransientFieldValueType(Class<V> valueClass) {
 
 		this.valueClass = Objects.requireNonNull(valueClass);
-		this.comparator = Optional.empty();
+		this.comparator = Optional.ofNullable(getNaturalOrderComparatorOrNull(valueClass));
```
Comment (alexander-schmidt) [inline platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/transients/TransientFieldValueType.java:15]: We now use a natural-order `Comparator` for `Comparable` value types, where we used none before.
Shouldn't this change have caused a test to fail?

Comment (oliver-richers) [reply platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/transients/TransientFieldValueType.java:15]: lots of tests are missing, so creating demo module and then tests

Comment (oliver-richers) [reply platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/transients/TransientFieldValueType.java:15]: as discussion, implemented new DemoModule and added tests for sorting and filtering transient field columns

**Comment by oliver-richers** [comment]
Sorting:
image

Filtering:
image

File: platform-demo-module/src/test/java/com/softicar/platform/demo/module/invoice/transients/sorting/DemoInvoicesTransientFieldFilteringTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package com.softicar.platform.demo.module.invoice.transients.sorting;
```
Comment (alexander-schmidt) [inline platform-demo-module/src/test/java/com/softicar/platform/demo/module/invoice/transients/sorting/DemoInvoicesTransientFieldFilteringTest.java:1]: Should this really reside in the `sorting` package?

Comment (oliver-richers) [reply platform-demo-module/src/test/java/com/softicar/platform/demo/module/invoice/transients/sorting/DemoInvoicesTransientFieldFilteringTest.java]: indeed, moved both test classes up

File: platform-demo-module/src/test/java/com/softicar/platform/demo/module/AbstractDemoModuleTest.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.softicar.platform.demo.module;
+
+import com.softicar.platform.core.module.user.AGUser;
+import com.softicar.platform.core.module.user.CurrentUser;
+import com.softicar.platform.db.runtime.test.AbstractDbTest;
+import com.softicar.platform.demo.module.test.fixture.DemoModuleTestFixtureMethods;
+
+public class AbstractDemoModuleTest extends AbstractDbTest implements DemoModuleTestFixtureMethods {
```
Comment (alexander-schmidt) [inline platform-demo-module/src/test/java/com/softicar/platform/demo/module/AbstractDemoModuleTest.java:8]: Should actually be `abstract`

File: platform-dom/src/main/java/com/softicar/platform/dom/document/marker/IDomDocumentMarkerHolder.java [hunk 4]
```diff
@@ -20,6 +20,8 @@
 
 	/**
 	 * Defines the marker for the given {@link IDomNode}.
+	 * <p>
+	 * TODO This method should be rather called <i>addMarker</i>.
```
Comment (alexander-schmidt) [inline platform-dom/src/main/java/com/softicar/platform/dom/document/marker/IDomDocumentMarkerHolder.java:24]: We should add a reference to `PLAT-568` here.

Comment (alexander-schmidt) [reply platform-dom/src/main/java/com/softicar/platform/dom/document/marker/IDomDocumentMarkerHolder.java:24]: Also, "marker" methods should be pluralized, i.e. `setMarker -> setMarkers`, `hasMarker -> hasMarkers`, `withMarker -> withMarkers`

File: platform-dom/src/main/java/com/softicar/platform/dom/elements/testing/node/tester/AbstractDomNodeTester.java [hunk 5]
```diff
@@ -231,11 +231,13 @@ public boolean containsNoText() {
 
 	// ------------------------------ retrieve textual content ------------------------------ //
 
+	// TODO this method should be renamed to getAllText (or similar)
 	public String getAllTextInDocument(String delimiter) {
 
 		return getAllTextsInTree().collect(Collectors.joining(delimiter));
 	}
 
+	// TODO this method should be renamed to getAllTextAsStream (or similar)
```
Comment (alexander-schmidt) [inline platform-dom/src/main/java/com/softicar/platform/dom/elements/testing/node/tester/AbstractDomNodeTester.java:240]: This method deserves some javadoc. Particularly, that should answer the question whether text in the node itself would be included in the result (as opposed to: only the texts in child nodes).

File: platform-dom/src/main/java/com/softicar/platform/dom/elements/testing/node/tester/AbstractDomNodeTester.java [hunk 6]
```diff
@@ -231,11 +231,13 @@ public boolean containsNoText() {
 
 	// ------------------------------ retrieve textual content ------------------------------ //
 
+	// TODO this method should be renamed to getAllText (or similar)
```
Comment (alexander-schmidt) [inline platform-dom/src/main/java/com/softicar/platform/dom/elements/testing/node/tester/AbstractDomNodeTester.java:234]: This method deserves some javadoc. Particularly, that should answer the question whether text in the node itself would be included in the result (as opposed to: only the texts in child nodes).

Comment (alexander-schmidt) [reply platform-dom/src/main/java/com/softicar/platform/dom/elements/testing/node/tester/AbstractDomNodeTester.java:234]: Also, `PLAT-568` should be referenced.

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

File: platform-demo-module/src/test/java/com/softicar/platform/demo/module/AbstractDemoModuleTest.java (hunk 3 above)
Comment (oliver-richers) [reply platform-demo-module/src/test/java/com/softicar/platform/demo/module/AbstractDemoModuleTest.java:8]: ok

File: platform-dom/src/main/java/com/softicar/platform/dom/document/marker/IDomDocumentMarkerHolder.java (hunk 4 above)
Comment (oliver-richers) [reply platform-dom/src/main/java/com/softicar/platform/dom/document/marker/IDomDocumentMarkerHolder.java:24]: ok

File: platform-dom/src/main/java/com/softicar/platform/dom/elements/testing/node/tester/AbstractDomNodeTester.java (hunk 5 above)
Comment (oliver-richers) [reply platform-dom/src/main/java/com/softicar/platform/dom/elements/testing/node/tester/AbstractDomNodeTester.java:240]: ok

File: platform-dom/src/main/java/com/softicar/platform/dom/elements/testing/node/tester/AbstractDomNodeTester.java (hunk 6 above)
Comment (oliver-richers) [reply platform-dom/src/main/java/com/softicar/platform/dom/elements/testing/node/tester/AbstractDomNodeTester.java:234]: ok

**Review by alexander-schmidt — APPROVED** [review]

**Pull request merged**